src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/OffenderService.kt

```diff
@@ -0,0 +1,102 @@
+class OffenderService(
+  private val communityApiClient: CommunityApiClient,
+  private val prisonsApiClient: PrisonsApiClient,
+) {
+
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  fun getInfoForPerson(crn: String, nomisUsername: String): PersonInfoResult {
+    var offenderResponse = communityApiClient.getOffenderDetailSummaryWithWait(crn)
+
+    if (offenderResponse is ClientResult.Failure.PreemptiveCacheTimeout) {
+      offenderResponse = communityApiClient.getOffenderDetailSummaryWithCall(crn)
+    }
+
+    val offender = when (offenderResponse) {
+      is ClientResult.Success -> offenderResponse.body
+
+      is ClientResult.Failure.StatusCode -> if (offenderResponse.status == HttpStatus.NOT_FOUND) {
+        return PersonInfoResult.NotFound(crn)
+      } else {
+        return PersonInfoResult.Unknown(crn, offenderResponse.toException())
+      }
+
+      is ClientResult.Failure -> return PersonInfoResult.Unknown(crn, offenderResponse.toException())
+    }
+
+    val inmateDetails = offender.otherIds.nomsNumber?.let { nomsNumber ->
+      when (val inmateDetailsResult = getInmateDetailByNomsNumber(offender.otherIds.crn, nomsNumber)) {
+        is AuthorisableActionResult.Success -> inmateDetailsResult.entity
+        else -> null
+      }
+    }
+
+    return PersonInfoResult.Success.Full(
+      crn = crn,
+      offenderDetailSummary = offender,
+      inmateDetail = inmateDetails,
+    )
+  }
+
+  fun getInmateDetailByNomsNumber(crn: String, nomsNumber: String): AuthorisableActionResult<InmateDetail?> {
+    var inmateDetailResponse = prisonsApiClient.getInmateDetailsWithWait(nomsNumber)
+
+    val hasCacheTimedOut = inmateDetailResponse is ClientResult.Failure.PreemptiveCacheTimeout
+    if (hasCacheTimedOut) {
+      inmateDetailResponse = prisonsApiClient.getInmateDetailsWithCall(nomsNumber)
+    }
+
+    fun logFailedResponse(inmateDetailResponse: ClientResult.Failure<InmateDetail>) = when (hasCacheTimedOut) {
+      true -> log.warn("Could not get inmate details for $crn after cache timed out", inmateDetailResponse.toException())
+      false -> log.warn("Could not get inmate details for $crn as an unsuccessful response was cached", inmateDetailResponse.toException())
+    }
+
+    val inmateDetail = when (inmateDetailResponse) {
+      is ClientResult.Success -> inmateDetailResponse.body
+      is ClientResult.Failure.StatusCode -> when (inmateDetailResponse.status) {
+        HttpStatus.NOT_FOUND -> return AuthorisableActionResult.NotFound()
+        HttpStatus.FORBIDDEN -> return AuthorisableActionResult.Unauthorised()
+        else -> {
+          logFailedResponse(inmateDetailResponse)
+          null
+        }
+      }
+      is ClientResult.Failure -> {
+        logFailedResponse(inmateDetailResponse)
+        null
+      }
+    }
+
+    return AuthorisableActionResult.Success(inmateDetail)
+  }
+
+  fun getOffenderByCrn(crn: String): AuthorisableActionResult<OffenderDetailSummary> {
```

Comment:
What's the thinking behind namespacing this in CAS2?

Reply:
CAS2 needs to be disentangle itself from Delius users and so we've reimplemented a number of functions which we need but without their Delius connection -- which is usually in the form of a test for LAO access against Delius' `offenders/crn/{crn}/userAccess` endpoint:

- `getOffenderbyCrn(crn: String, userDistinguishedName: String, ignoreLao: Boolean = false)`
- `getInfoForPerson(crn: String, deliusUsername: String, ignoreLao: Boolean)`
- `getRiskByCrn(crn: String, jwt: String, userDistinguishedName: String)` In this function we also skip the upstream API calls to fetch Mappa, Tier and Flag info which is not needed

We do, however, continue to use the original `OffenderService` where possible, e.g. to get OASys information via:

- `OffenderService.getOASysOffenceDetails(crn)`
- `OffenderService.getOASysRiskToTheIndividual(crn)`
- `OffenderService.getOASysRoshSummary(crn)`

We bring the original OffenderService into our CAS2 PeopleController with:

`import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.OffenderService as OASysOffenderService`